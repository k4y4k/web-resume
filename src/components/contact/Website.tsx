import * as React from 'react'
import { FiGlobe } from '@react-icons/all-files/fi/FiGlobe'

interface WebsiteTypes {
  url: string
}

const Website = ({ url }: WebsiteTypes): JSX.Element | null => {
  if (url === '') return null

  const stripped = url?.replace(/https?:\/\//i, '')

  return (
    <a data-testid='contactWebsite' href={url} target='_blank' rel='noreferrer'>
      <FiGlobe /> {stripped}
    </a>
  )
}

export default Website
