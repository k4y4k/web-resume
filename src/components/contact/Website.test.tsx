import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Website from './Website'

describe('<Website />', () => {
  test('handles no data', () => {
    render(<Website url='' />)

    const website = screen.queryByTestId('contactWebsite')
    expect(website).not.toBeInTheDocument()
    expect(website).toMatchSnapshot()
  })

  test('strips out protocol to display website', () => {
    render(<Website url='https://example.com' />)

    const website = screen.getByTestId('contactWebsite')
    expect(website).toHaveTextContent('example.com')
  })

  test('links to website in new tab', () => {
    render(<Website url='https://example.com' />)

    const website = screen.getByTestId('contactWebsite')
    expect(website).toHaveAttribute('href', 'https://example.com')
    expect(website).toHaveAttribute('target', '_blank')
    expect(website).toMatchSnapshot()
  })
})
