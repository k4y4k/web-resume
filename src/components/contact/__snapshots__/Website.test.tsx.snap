// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Website /> handles no data 1`] = `null`;

exports[`<Website /> links to website in new tab 1`] = `
<a
  data-testid="contactWebsite"
  href="https://example.com"
  rel="noreferrer"
  target="_blank"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
    />
    <line
      x1="2"
      x2="22"
      y1="12"
      y2="12"
    />
    <path
      d="M12 2a15.3 15.3 0 0 1 4 10 15.3 15.3 0 0 1-4 10 15.3 15.3 0 0 1-4-10 15.3 15.3 0 0 1 4-10z"
    />
  </svg>
   
  example.com
</a>
`;
